monitor/monitor.go

```diff
@@ -58,7 +56,15 @@ func (m *Monitor) Run(npages int) {
 	}
 }
 
+func setNumPages(npages int) {
```

Comment:
Create a method for `PerfEventConfig` with a private mutex to set the `NumPages`